core/sdk-core/src/main/java/software/amazon/awssdk/core/client/builder/SdkDefaultClientBuilder.java

```diff
@@ -353,10 +353,14 @@ private ScheduledExecutorService resolveScheduledExecutorService() {
 
     @Override
     public final B endpointOverride(URI endpointOverride) {
-        Validate.paramNotNull(endpointOverride, "endpointOverride");
-        Validate.paramNotNull(endpointOverride.getScheme(), "The URI scheme of endpointOverride");
-        clientConfiguration.option(SdkClientOption.ENDPOINT, endpointOverride);
-        clientConfiguration.option(SdkClientOption.ENDPOINT_OVERRIDDEN, true);
```

Comment:
Why are we removing the null check for endpointoverride?

Reply:
Because we're now handling null as "reset to default". Essentially, endpointOverride won't ever be null in this statement block.